src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -288,8 +293,10 @@ of depth `k`.
 function BinaryTree(k::T) where {T <: Integer}
     k <= 0 && return SimpleGraph(0)
     k == 1 && return SimpleGraph(1)
-    Tw = widen(T)
-    n = T(Tw(2)^Tw(k) - Tw(1))  # checks if T is large enough for 2^k
+    if !(isa(k, BigInt) || BigInt(2) ^ k - 1 <= typemax(k)) # TODO check for other possible infinite Integer types
+        throw(DomainError(k, "2^k - 1 not representable by type $T"))
+    end
+    n = T(2^k - 1)
```

Comment:
This is way too complicated and relies on `isa` which is a code smell. What's wrong with the `widen` code?

Reply:
I think I could just write a method
```
isbounded(n::Integer) = true
isbounded(n::BigInt) = false
```
and use that instead of `isa`. Would that be better?

Reply:
yeah then you get a method error with the suggestions for Integer and BigInt in the stacktrace.